flink-runtime/src/main/java/org/apache/flink/runtime/security/token/hadoop/KerberosLoginProvider.java

```diff
@@ -99,6 +101,8 @@ public void doLogin() throws IOException {
             LOG.info("Attempting to load user's ticket cache");
             UserGroupInformation.loginUserFromSubject(null);
             LOG.info("Loaded user's ticket cache successfully");
+        } else if (supportProxyUser) {
+            LOG.info("Delegation token fetch is managed and therefore login is managed");
```

Comment:
Maybe we can say something like: `Proxy user doesn't need login since it must have credentials already`. I think `KerberosLoginProvider` shouldn't know about tokens at all.